### ISSUE TITLE
Subscription Form block: Make the confirmation message visible after submitting the form

### ISSUE THREAD

**Issue by AtrumGeost** [Automattic/jetpack#18015]
#### Is your feature request related to a problem? Please describe.

Yes, here's how to replicate it:

1. Add the Subscription Form Block at the bottom of a page (long enough you need to scroll down)
2. On the front end try subscribing. 
3. Notice after clicking on submit, the page reloads and you are lead to the top of the page. You would need to scroll down again to see the confirmation message.

Here's a video of what happens:

https://d.pr/v/MZVssz/kEqn2LoyB4

#### Describe the solution you'd like

To not to have to scroll down to see the subscription form message. 

#### Additional context

User report: 26105851-hc

**Comment by andres-blanco** [comment]
**Progress**

We could add a custom class when the notification appears and scroll via javascript doing something like this:
```js
document.getElementsByClassName('sub-notification')[0].scrollIntoView()
```

I will find out the best way to include that JS snippet

**Comment by kwight** [comment]
It would be nice if the screen didn't move at all, but I think we'd need to change how the whole block works; it looks like it uses the Subscription widget on the front-end (so the button is a form submission).

<img width="1533" alt="Screen Shot 2021-08-17 at 1 31 17 PM" src="https://user-images.githubusercontent.com/349751/129796577-d4868a86-677d-40c4-a19b-92c4e7f4daa7.png">

True, `scrollIntoView` with the default behaviour could be an acceptable compromise (janky experience, but effective).

**Comment by andres-blanco** [comment]
yes, the block submits a form to `subscribe.wordpress.com`. I think this is a good compromise considering how much the block would need to change in order to make a cleaner fix.

I created a diff here D65642-code
